### PR TITLE
4.x: HTTP2 concurrent streams check

### LINKED ISSUE #7695: 4.x: HTTP/2 Closed streams are not ditched
Streams ended with endStream flag from the server side are not removed from connection stream map. Causes `SETTINGS_MAX_CONCURRENT_STREAMS` validation to send GOAWAY for connections who behaves properly.

### PR DESCRIPTION
**Pull request by danielkec**
Fixes #7695

### CODE REVIEW

File: webserver/http2/src/main/java/io/helidon/webserver/http2/Http2ServerStream.java [hunk 1]
```diff
@@ -114,7 +116,8 @@ public class Http2ServerStream implements Runnable, Http2Stream {
      * @param writer                writer
      * @param connectionFlowControl connection flow control
      */
-    public Http2ServerStream(ConnectionContext ctx,
+    Http2ServerStream(ConnectionContext ctx,
```
Comment (tomas-langer) [inline webserver/http2/src/main/java/io/helidon/webserver/http2/Http2ServerStream.java:119]: This deserves reformatting of parameters, as the `public` was removed

File: http/http2/src/main/java/io/helidon/http/http2/Http2ConnectionWriter.java [hunk 2]
```diff
@@ -164,27 +170,26 @@ public int writeHeaders(Http2Headers headers,
...
             noLockWrite(frame);
-            return null;
-        });
+        } finally {
+            streamLock.unlock();
+        }
     }
 
-    private <T> T withStreamLock(Callable<T> callable) {
+    private void lock() {
```
Comment (tomas-langer) [inline http/http2/src/main/java/io/helidon/http/http2/Http2ConnectionWriter.java:190]: Why is this refactored?

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: http/http2/src/main/java/io/helidon/http/http2/Http2ConnectionWriter.java (hunk 2 above)
Comment (zimmi) [reply http/http2/src/main/java/io/helidon/http/http2/Http2ConnectionWriter.java:190]: Hi, please excuse the comment from the peanut gallery, but I recently stumbled over the rationale why such a method wasn't added to the `Lock` interface itself: https://bugs.openjdk.org/browse/JDK-8025597?focusedCommentId=13414177&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13414177

**Review by zimmi — APPROVED** [review]
Looks great :)

File: webserver/http2/src/main/java/io/helidon/webserver/http2/Http2ServerStream.java (hunk 1 above)
Comment (danielkec) [reply webserver/http2/src/main/java/io/helidon/webserver/http2/Http2ServerStream.java:119]: fixed

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**